cns/middlewares/SWIFTv2.go

```diff
@@ -0,0 +1,111 @@
+package middlewares
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/cns/types"
+	"github.com/Azure/azure-container-networking/crd/multitenancy/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ErrMTPNCNotReady = errors.New("mtpnc is not ready")
+
+type SWIFTv2Middleware struct {
+	// TODO: implement
+	// need cached scoped client for pods
+	// need client for MTPNC CRD for x-ref pods
+	cli client.Client
+}
+
+func NewSWIFTv2Middleware(cli client.Client) *SWIFTv2Middleware {
+	return &SWIFTv2Middleware{
+		cli: cli,
+	}
+}
+
+// Return the validator function for the middleware
+func (m *SWIFTv2Middleware) Validator() cns.IPConfigValidator {
+	return m.validateMultitenantIPConfigsRequest
+}
+
+// validateMultitenantIPConfigsRequest validates if pod is multitenant
+// nolint
+func (m *SWIFTv2Middleware) validateMultitenantIPConfigsRequest(req *cns.IPConfigsRequest) (respCode types.ResponseCode, message string) {
+	// Retrieve the pod from the cluster
+	podInfo, err := cns.UnmarshalPodInfo(req.OrchestratorContext)
+	if err != nil {
+		errBuf := fmt.Sprintf("unmarshalling pod info from ipconfigs request %v failed with error %v", req, err)
+		return types.UnexpectedError, errBuf
+	}
+	podNamespacedName := k8stypes.NamespacedName{Namespace: podInfo.Namespace(), Name: podInfo.Name()}
+	pod := v1.Pod{}
+	err = m.cli.Get(context.TODO(), podNamespacedName, &pod)
```

Comment:
let's pass a context in (and thread it back up the call stack as far as necessary)

Reply:
we are not passing down any context in the current RequestIPConfigs flow, how far up the call stack are we talking about?

Reply:
At least up to the http.Handler - the context should exist in the scope of a single request tx

Reply:
would it make more sense to open another PR for this? Since we have to do refractor in other places as well that are not related to this PR

Reply:
that's fine